sympy/matrices/tests/test_commonmatrix.py

```diff
@@ -1059,3 +1060,8 @@ class Foo(ImmutableDenseMatrix):
 
     assert isinstance(c, Foo)
     assert c == Matrix([[7, 10], [15, 22]])
+
+def test_issue_18956():
+    A = Array([[1, 2], [3, 4]])
+    B = Matrix([[1,2],[3,4]])
+    raises(TypeError, lambda: B + A)
```

Comment:
This should test `A+B` as well as `B+A`

Reply:
Oh I see that you did that in the other file. The tests should be combined in a single function.

Reply:
I have added A + B , but that test shouldn't be here in this file because that would test Array class not matrix, that's why I added that test case to test_ndim_array.py

Reply:
Both tests are for the interaction between Array and Matrix. They should go together.

Reply:
But
Matrix + array would invoke __add__() of class Matrix , so it is in Matrix test
Array + Matrix would invoke __add__() of class Array, so it is in Array test 

How can they go together?

Reply:
@oscarbenjamin shall both test case be added in a same file or not?